src/ray/object_manager/object_manager_client_connection.h

```diff
@@ -7,63 +7,141 @@
 
 #include <boost/asio.hpp>
 #include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
 #include <boost/enable_shared_from_this.hpp>
 
+#include "common/state/ray_config.h"
+#include "ray/common/client_connection.h"
 #include "ray/id.h"
 
 namespace ray {
 
-struct SendRequest {
-  ObjectID object_id;
-  ClientID client_id;
-  int64_t object_size;
-  uint8_t *data;
-};
-
-// TODO(hme): Document public API after integration with common connection.
-class SenderConnection : public boost::enable_shared_from_this<SenderConnection> {
+class SenderConnection : public ServerConnection<boost::asio::ip::tcp>,
```

Comment:
What is preventing you from using `ServerConnection` directly? We should really try to avoid patterns like this where we have to subclass just to add very limited functionality that's specific to one use case. Instead, embed the `ServerConnection` inside of a class specific to the object manager.

Reply:
Please see comment in #1791 on implementing this refactor while maintaining multi-threading requirements.